test/studies/arkouda/sub_test

```diff
@@ -81,6 +81,11 @@ if [ -z "${CHPL_TEST_ARKOUDA_SKIP_CHAPEL_PY}" ] ; then
   (cd $CHPL_HOME && make chapel-py-venv)
 fi
 
+# Enable multi dim array support
+if [ -n "${CHPL_TEST_ARKOUDA_MULTI_DIM}" ] ; then
+  apply_git_patch $CHPL_HOME/test/studies/arkouda/multidim.patch $ARKOUDA_HOME
```

Comment:
I am not too opposed to this approach to get the ball rolling. But looking around a bit more in the Makefile and the register_commands.py, I see that we can use `ARKOUDA_REGISTRATION_CONFIG` environment to specify a custom file instead of `registration-config.json`. Using that approach feels more maintainable than patching the repo to me.

Reply:
`registration-config.json` could be modified to add more types, dtypes, etc. I feel like if we specified `ARKOUDA_REGISTRATION_CONFIG` in that fashion, it would be easy to let changes like that quietly fly under the radar and then we won't be testing what we think we are since the version we have in this repo won't be updated.

I like this better because if something is changed in `registration-config`, we get the chance to make a decision about what to do when things break.

Reply:
I agree with Shreyas here, this approach is less likely to get out of sync with the Arkouda main.

Reply:
If @e-kayrakli doesn't object, I'll merge this as is today.

Reply:
Yeah, I understand your worry there. I still have a wish for using a user-facing way instead of patching the code base. But maybe the ideal solution is to have a command line flag for the `register_commands.py` script?

I am OK with merging this as is.

Reply:
Allowing command line flags for register_commands seems very attractive to me.

Reply:
Yes, I'm in favor of that too. If a flag can allow us to specify the dimensions we want to build arkouda with, it makes life better for users too.

Reply:
Note that there is `MaxArrayDims` that's somewhat similar to what we are discussing here. It'd be good to avoid creating a confusing situation with two compile-time knobs about dimensionality. FWIW, [here](https://github.com/Bears-R-Us/arkouda/pull/4204#issuecomment-2751002234) we were discussing to remove it.

Reply:
I created https://github.com/Bears-R-Us/arkouda/issues/4225

Reply:
I'd also argue that having a separate registration-config.json and a ServerModules.ini file is confusing, and  combining the two is ideal. i.e. why do we have two files to configure the server at compile time?

Reply:
I don't know how on-board I am with combining the two files, but the reason they are different is probably because Arkouda was modular from the beginning but the function registration framework was added later and it used a different file.